merlin/models/tf/core.py

```diff
@@ -2683,8 +2823,8 @@ def load_topk_evaluation(self, data: merlin.io.Dataset, k: int, **kwargs):
             self.retrieval_block.item_block(), data=data, k=k - 1, context=self.context, **kwargs
         )
         # set cache_query to True in the ItemRetrievalScorer
-        self.loss_block.retrieval_scorer.cache_query = True
-        self.loss_block.pre_metrics = topk_index
+        self.loss_block.set_retrieval_cache_query(True)
```

Comment:
I think we should we check whether these functions are defined in the loss_block first. [This](https://github.com/NVIDIA-Merlin/models/blob/65e78feaac37fcc831de1fc7d043e4eb1f4751f9/merlin/models/tf/utils/mixins.py#L98) is the type, so it could not be there.

Reply:
Now I am checking if `self.loss_block` is defined and if it is an instance of `ItemRetrievalTask`